web/src/lib/services/shopping.ts

```diff
@@ -0,0 +1,95 @@
+import type {ShoppingCartItem} from "../types/nip45";
+import {Error, Info, ShoppingCart} from "../stores";
+import productImageFallback from "$lib/images/product_image_fallback.svg";
+
+// =============================== Products ====================================
+export function onQtyChangeClick(event, plus = false) {
+    let parentOfClickedButton = event.srcElement.parentElement.parentElement.parentElement;
+
+    const quantitySelector = parentOfClickedButton.getElementsByClassName('quantitySelector');
+
+    if (quantitySelector.length !== 1) {
+        Error.set('There was an error trying to modify the quantity.');
+        return;
+    }
+
+    let actualValue = Number(quantitySelector[0].value);
+
+    let newValue = plus ? ++actualValue : --actualValue;
+
+    if (newValue === 0 && plus === false) {
+        return;
+    }
+
+    quantitySelector[0].value = newValue;
+}
+
+export function onImgError(image) {
+    image.onerror = "";
+    image.src = productImageFallback;
+}
+
+// =============================== Shopping Cart ===============================
+export function addToCart(added_product: ShoppingCartItem, event) {
+    // console.log('Adding to cart: ', product, event);
+    const addToCartButton = event.srcElement;
+
+    const quantitySelector = addToCartButton.parentNode.getElementsByClassName('quantitySelector');
+    if (quantitySelector.length !== 1) {
+        Error.set('There was an error trying to get the amount you want to buy.');
+        return;
+    }
+
+    const quantitySelectorValue = Number(quantitySelector[0].value);
```

Comment:
Would be nicer if the quantity selector just exposed the count using "export function getQuantity() { return quantity; }" and we would get a reference to the component and do `quantityEditor.getQuantity()` no? Rather than relying on navigating the DOM.